ras_backstage/resources/sample/sample.py

```diff
@@ -5,32 +5,33 @@
 from structlog import wrap_logger
 
 from ras_backstage import sample_api
-from ras_backstage.common.filters import get_collection_exercise_by_period
 from ras_backstage.controllers import collection_exercise_controller, sample_controller, survey_controller
 
 
 logger = wrap_logger(logging.getLogger(__name__))
 
 parser = reqparse.RequestParser()
 parser.add_argument('file', location='files', required=True)
+parser.add_argument('collection_exercise_id', location='args', required=True)
 
 
-@sample_api.route('/<short_name>/<period>')
+@sample_api.route('')
 class Sample(Resource):
 
     @staticmethod
     @sample_api.expect(parser)
-    def post(short_name, period):
-        logger.info('Uploading sample', shortname=short_name, period=period)
-        survey = survey_controller.get_survey_by_shortname(short_name)
+    def post():
+        collection_exercise_id = request.args.get('collection_exercise_id')
+        if not collection_exercise_id:
+            return make_response(jsonify({'message': 'Missing collection_exercise_id'}), 400)
 
-        exercises = collection_exercise_controller.get_collection_exercises_by_survey(survey['id'])
-        # Find the collection exercise for the given period
-        exercise = get_collection_exercise_by_period(exercises, period)
+        logger.info('Uploading sample', collection_exercise_id=collection_exercise_id)
+
+        exercise = collection_exercise_controller.get_collection_exercise_by_id(collection_exercise_id)
```

Comment:
is this check needed? does sample service not have the validation in?

Reply:
@tej99 
The check may not be needed but the sample service has no knowledge of the collection exercise that the sample is being uploaded against (until, presumably, the file is processed). Perhaps it's worth guarding against uploading a sample that is for a currently unknown CE?

If we do hand over validation to the sample service, the sample upload resource probably doesn't need any parameters – survey short name, collection exercise or otherwise.